src/passes/Asyncify.cpp

```diff
@@ -1316,6 +1344,80 @@ struct AsyncifyAssertInNonInstrumented : public Pass {
   Module* module;
 };
 
+struct AsyncifyAssertUnwindCorrectness : Pass {
+  ModuleAnalyzer* analyzer;
+  Module* module;
+
+  AsyncifyAssertUnwindCorrectness(ModuleAnalyzer* analyzer, Module* module) {
+    this->analyzer = analyzer;
+    this->module = module;
+  }
+
+  std::unique_ptr<Pass> create() override {
+    return std::make_unique<AsyncifyAssertUnwindCorrectness>(analyzer, module);
+  }
+
+  bool isFunctionParallel() override { return true; }
+
+  void runOnFunction(Module*, Function* function) override {
+    auto builder = std::make_unique<Builder>(*module);
```

Comment:
Builders are very cheap to construct - please create one in `replaceCallWithCheck`, as needed, which would be simpler.